### ISSUE TITLE
compilation error

### ISSUE THREAD

**Issue by kkonevets** [dylanowen/mdbook-graphviz#21]
I run `cargo install mdbook-graphviz`

```
error[E0308]: mismatched types
  --> /home/guyos/.cargo/registry/src/github.com-1ecc6299db9ec823/mdbook-graphviz-0.1.3/src/preprocessor.rs:87:25
   |
86 |                     match e {
   |                           - this expression has type `pulldown_cmark::Event<'_>`
87 |                         Event::Text(ref text) => {
   |                         ^^^^^^^^^^^^^^^^^^^^^ expected enum `pulldown_cmark::Event`, found enum `Event`

error[E0308]: mismatched types
  --> /home/guyos/.cargo/registry/src/github.com-1ecc6299db9ec823/mdbook-graphviz-0.1.3/src/preprocessor.rs:93:25
   |
86 |                     match e {
   |                           - this expression has type `pulldown_cmark::Event<'_>`
...
93 |                         Event::End(Tag::CodeBlock(CodeBlockKind::Fenced(ref info_string))) => {
   |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `pulldown_cmark::Event`, found enum `Event`

error[E0308]: mismatched types
   --> /home/guyos/.cargo/registry/src/github.com-1ecc6299db9ec823/mdbook-graphviz-0.1.3/src/preprocessor.rs:114:25
    |
113 |                     match e {
    |                           - this expression has type `pulldown_cmark::Event<'_>`
114 |                         Event::Start(Tag::CodeBlock(CodeBlockKind::Fenced(ref info_string)))
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `pulldown_cmark::Event`, found enum `Event`

error[E0308]: `if` and `else` have incompatible types
   --> /home/guyos/.cargo/registry/src/github.com-1ecc6299db9ec823/mdbook-graphviz-0.1.3/src/preprocessor.rs:113:21
    |
85  |   /                 if let Some(mut builder) = graphviz_block_builder.take() {
86  |   |                     match e {
    |  _|_____________________-
87  | | |                         Event::Text(ref text) => {
88  | | |                             builder.append_code(&**text);
89  | | |                             graphviz_block_builder = Some(builder);
...   | |
110 | | |                         }
111 | | |                     }
    | |_|_____________________- expected because of this
112 |   |                 } else {
113 | / |                     match e {
114 | | |                         Event::Start(Tag::CodeBlock(CodeBlockKind::Fenced(ref info_string)))
115 | | |                             if (&**info_string).find(INFO_STRING_PREFIX) == Some(0) =>
116 | | |                         {
...   | |
125 | | |                         _ => Ok(vec![e]),
126 | | |                     }
    | |_|_____________________^ expected enum `Event`, found enum `pulldown_cmark::Event`
127 |   |                 }
    |   |_________________- `if` and `else` have incompatible types
    |
    = note: expected type `Result<Vec<Event<'_>>, mdbook::errors::Error>`
               found enum `Result<Vec<pulldown_cmark::Event<'_>>, _>`

For more information about this error, try `rustc --explain E0308`.
error: failed to compile `mdbook-graphviz v0.1.3`, intermediate artifacts can be found at `/tmp/cargo-installhWTwcT`

Caused by:
  could not compile `mdbook-graphviz` due to 4 previous errors
```

**Comment by EQt** [comment]
```bash
$ cargo install --git=https://github.com/dylanowen/mdbook-graphviz.git
```
works for me. I think it is related to outdated version of `pulldown_cmark` which was fixed in pull request #19 .

**Comment by dylanowen** [comment]
Just published https://github.com/dylanowen/mdbook-graphviz/releases/tag/v0.1.4 which should fix this issue, let me know if you're still having any problems
